router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go

```diff
@@ -0,0 +1,606 @@
+	destinationIDFileMap map[string]string
+	timeout              time.Duration
+}
+
+type User struct {
+	Email       string `json:"email"`
+	HashedEmail string `json:"hashedEmail"`
+}
+type Message struct {
+	List []User `json:"List"`
+}
+type Metadata struct {
+	JobID int64 `json:"job_id"`
+}
+
+// This struct represent each line of the text file created by the batchrouter
+type Data struct {
+	Message  Message  `json:"message"`
+	Metadata Metadata `json:"metadata"`
+}
+
+type DestinationConfig struct {
+	AudienceId               string   `json:"audienceId"`
+	CustomerAccountId        string   `json:"customerAccountId"`
+	CustomerId               string   `json:"customerId"`
+	OneTrustCookieCategories []string `json:"oneTrustCookieCategories"`
+	RudderAccountId          string   `json:"rudderAccountId"`
+}
+
+type secretStruct struct {
+	AccessToken     string
+	RefreshToken    string
+	Developer_token string
+	ExpirationDate  string
+}
+
+type Response struct {
+	Status   string      `json:"status"`
+	Metadata MetadataNew `json:"metadata"`
+}
+
+type MetadataNew struct {
+	FailedKeys    []int64           `json:"failedKeys"`
+	FailedReasons map[string]string `json:"failedReasons"`
+	WarningKeys   []string          `json:"warningKeys"`
+	SucceededKeys []int64           `json:"succeededKeys"`
+}
+
+/*
+This function create zip file from the text file created by the batchrouter
+It takes the text file path as input and returns the zip file path
+The maximum size of the zip file is 100MB, if the size of the zip file exceeds 100MB then the job is marked as failed
+*/
+func createZipFile(filePath string, failedJobIds *[]int64, successJobIds *[]int64, audienceId string) string {
+
+	localTmpDirName := fmt.Sprintf(`/%s/`, misc.RudderAsyncDestinationLogs)
+	uuid := uuid.New()
+	tmpDirPath, err := misc.CreateTMPDIR()
+	if err != nil {
+		panic(err)
+	}
+	path := fmt.Sprintf("%v%v", tmpDirPath+localTmpDirName, fmt.Sprintf("%v", uuid.String()))
+	csvFilePath := fmt.Sprintf(`%v.csv`, path)
+	zipFilePath := fmt.Sprintf(`%v.zip`, path)
+	textFile, err := os.Open(filePath)
+	if err != nil {
+		panic(err)
+	}
+	defer textFile.Close()
+	csvFile, err := os.Create(csvFilePath)
+	if err != nil {
+		panic(err)
+	}
+	csvWriter := csv.NewWriter(csvFile)
+	csvWriter.Write([]string{"Type", "Status", "Id", "Parent Id", "Client Id", "Modified Time", "Name", "Description", "Scope", "Audience", "Action Type", "Sub Type", "Text"})
+	csvWriter.Write([]string{"Format Version", "", "", "", "", "", "6.0", "", "", "", "", "", ""})
+	csvWriter.Write([]string{"Customer List", "", audienceId, "", "", "", "", "", "", "", "Add", "", ""})
+	scanner := bufio.NewScanner(textFile)
+	size := 0
+	for scanner.Scan() {
+		line := scanner.Text()
+		var data Data
+		err := json.Unmarshal([]byte(line), &data)
+		if err != nil {
+			panic(err)
+		}
+		marshaledUploadlist, err := json.Marshal(data.Message.List)
+		size = size + len([]byte(marshaledUploadlist))
+		if size < 104857600 {
```

Comment:
It a good practice to avoid magic numbers put in the code like this, lets's create a constant or a configurable value for this.

 We might also want to use a multiple of an SI constant for readability https://github.com/rudderlabs/rudder-go-kit/blob/main/bytesize/bytesize.go

Reply:
@ItsSudip